migrations/manual/educandu-2021-12-16-01-make-cdn-urls-generic-in-annotations.js

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable no-await-in-loop, no-console */
+
+async function updateAll(collection, query, updateFn) {
+  const cursor = collection.find(query);
+
+  while (await cursor.hasNext()) {
+    const doc = await cursor.next();
+    const updatedDoc = await updateFn(doc);
+    if (updatedDoc) {
+      await collection.replaceOne({ _id: doc._id }, updatedDoc);
+    }
+  }
+}
+
+export function processSection(section) {
+  let updateCount = 0;
+  if (section.type === 'annotation' && section.content) {
+    const regexp = /\]\((https:\/\/((staging\.)?cdn\.elmu\.online)|(https:\/\/cdn\.((integration|staging)\.)?openmusic\.academy))\//g;
+    section.content.text = section.content.text.replace(regexp, () => {
+      updateCount += 1;
+      return '](cdn://';
+    });
+  }
+
+  return updateCount;
+}
+
+// eslint-disable-next-line camelcase
+export default class Educandu_2021_12_16_01_make_cdn_urls_generic_in_annotations {
+  constructor(db) {
+    this.db = db;
+  }
+
+  async up() {
+    await updateAll(this.db.collection('documentRevisions'), { 'sections.type': 'annotation' }, doc => {
+      const updateCount = doc.sections.reduce((accu, section) => accu + processSection(section), 0);
```

Comment:
I think the annotations were never moved to the cdn resources, we need to do that here too. The other script was relying on the fact that  2021-12-14-01 would be run.
Maybe we want to keep it into 2 separate things though. Just a remark to not lose sight.